src/main/java/org/radarcns/monitor/DisconnectMonitor.java

```diff
@@ -150,12 +217,77 @@ public void setLastSeen(Map<String, Long> lastSeen) {
             this.lastSeen.putAll(lastSeen);
         }
 
-        public Map<String, Long> getReportedMissing() {
+        public Map<String, String> getReportedMissing() {
             return reportedMissing;
         }
 
-        public void setReportedMissing(Map<String, Long> reportedMissing) {
+        public void setReportedMissing(Map<String, String> reportedMissing) {
             this.reportedMissing.putAll(reportedMissing);
         }
     }
+
+    /**
+     * Stores data of data from missing records alert
+     * such as lastSeen and reportedTime
+     */
+    public static class MissingRecordsReport {
+
+        private Long lastSeen;
+        private Long reportedMissing;
+
+        public MissingRecordsReport() {
+            //default constructor for ObjectMapper
+        }
+
+        @JsonCreator
+        public MissingRecordsReport(@JsonProperty("lastSeen") Long lastSeen,
+                @JsonProperty("reportedMissing") Long reportedMissing) {
+            this.lastSeen = lastSeen;
+            this.reportedMissing = reportedMissing;
+        }
+
+        public Long getLastSeen() {
+            return lastSeen;
+        }
+
+        public Long getReportedMissing() {
+            return reportedMissing;
+        }
+
+
+    }
+
+    /**
+     * Task that runs with scheduled delays to report if the device is still disconnected
+     */
+    private class ScheduledAlertsUpdates implements Runnable {
+
+        private final Logger logger = LoggerFactory.getLogger(ScheduledAlertsUpdates.class);
+
+        @Override
+        public void run() {
+
+            long now = System.currentTimeMillis();
+            logger.info("Scheduled alert updates at " + now);
+            Iterator<Map.Entry<String, String>> iterator = state.reportedMissing.entrySet()
+                    .iterator();
+            while (iterator.hasNext()) {
+                Map.Entry<String, String> entry = iterator.next();
+                MissingRecordsReport missingRecordsReport = null;
+                try {
+                    missingRecordsReport = stringToMissingRecordsReport(entry.getValue());
+                    long timeout = now - missingRecordsReport.getLastSeen();
+                    if (timeout > timeUntilReportedMissing) {
+                        String missingKey = entry.getKey();
+                        reportMissing(stringToKey(missingKey), timeout);
+                        state.reportedMissing.put(missingKey, missingRecordsReportToString(
+                                new MissingRecordsReport(missingRecordsReport.getLastSeen(), now)));
+                    }
+                } catch (IOException e) {
+                    logger.error("Cannot serialize/deserialize missing records data");
+                }
+
+            }
+        }
+    }
```

Comment:
Note that with this setup, all reminders will be sent in one batch and they will not expire. It would be nicer to have

1. a maximum number of messages sent
2. have the next message be scheduled based on when the timeout occurred.

The second step would require a bit more logic during start of the monitor, to restart all runnables that need to be re-sent. Could you add a TODO and/or Javadoc to make this clear?